### ISSUE TITLE
Onboarding Microsoft.Capacity RP

### ISSUE THREAD

**Issue by rahuls-microsoft** [Azure/azure-cli#17339]
**Resource Provider**

RP: Microsoft.Capacity
Features: reservations, Quota
**Description of Feature or Work Requested**

Details of Quota APIs:  https://docs.microsoft.com/en-us/rest/api/reserved-vm-instances/quotaapi

https://github.com/AzureSDKAutomation/azure-cli-extensions/pull/700 - This PR was created automatically. It was tested. How can we release it or take it to release?
**Minimum API Version Required**

"version": "2020-10-25",
Tag: package-2020-10-25

**Swagger Link**
https://github.com/Azure/azure-rest-api-specs/pull/13202

**Target Date**
3/31/2021

**Comment by yonzhan** [comment]
Capacity

**Comment by rahuls-microsoft** [comment]
Yes, Microsoft.Capacity RP. 
Will it generate PowerShell modules also or there is a separate process for it?
What's the timeline, we are already in GA and this has been requested by customers and ARM team.

**Comment by yonzhan** [comment]
@qiaozha Do you have latest status for this?

**Comment by qiaozha** [comment]
> @qiaozha Do you have latest status for this?

No, we haven't validated for this RP as they are not PLR and there's no need to because Service Team already had some tests on the generated cli by themselves.

**Comment by rahuls-microsoft** [comment]
Is there anything required from my side?
https://github.com/AzureSDKAutomation/azure-cli-extensions/pull/700 - was created automatically based on the swagger PR - https://github.com/Azure/azure-rest-api-specs/pull/13202

Please let me know, if any action is needed from myside?

**Comment by jsntcy** [comment]
@rahuls-microsoft,
For CLI, you need create a PR to the master branch in https://github.com/Azure/azure-cli-extensions, and we'll help review your PR.
For PowerShell, you need contact @dingmeng-xue.

**Comment by rahuls-microsoft** [comment]
> @rahuls-microsoft,
> For CLI, you need create a PR to the master branch in https://github.com/Azure/azure-cli-extensions, and we'll help review your PR.
> For PowerShell, you need contact @dingmeng-xue.

Hi, What do I need to add in the PR?
Copy the files generated from- https://github.com/AzureSDKAutomation/azure-cli-extensions/pull/700

Please guide me through the process.
Thanks.
Rahul.

**Comment by qiaozha** [comment]
You can cherry-pick the commits https://github.com/AzureSDKAutomation/azure-cli-extensions/pull/700/commits 

Or you can run `autorest --az` in your local then submit those generated files after you make sure `azdev test <your-module-name> --live` passes.

**Comment by rahuls-microsoft** [comment]
https://github.com/Azure/azure-cli-extensions/pull/3194 - I have created the PR based on - https://github.com/AzureSDKAutomation/azure-cli-extensions/pull/700/commits

**Comment by rahuls-microsoft** [comment]
Hi
I herry-pick the commits https://github.com/AzureSDKAutomation/azure-cli-extensions/pull/700/commits<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzureSDKAutomation%2Fazure-cli-extensions%2Fpull%2F700%2Fcommits&data=04%7C01%7Crahul.singh%40microsoft.com%7Cbfbcc52447dd4468ae0d08d8ef4ad726%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637522452048851186%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=K%2BfsCoTRCzHzaJSZOdOkwyYzrB0jrA%2BGH37Y%2FZiKD3s%3D&reserved=0> and created PR-Reservation and Quota CLI extension by rahuls-microsoft · Pull Request #3194 · Azure/azure-cli-extensions (github.com)<https://github.com/Azure/azure-cli-extensions/pull/3194>
[https://avatars.githubusercontent.com/u/6844498?s=400&v=4]<https://github.com/Azure/azure-cli-extensions/pull/3194>
Reservation and Quota CLI extension by rahuls-microsoft · Pull Request #3194 · Azure/azure-cli-extensions<https://github.com/Azure/azure-cli-extensions/pull/3194>
Copied from : https://github.com/AzureSDKAutomation/azure-cli-extensions/pull/700/commits This checklist is used to make sure that common guidelines for a pull request are followed. General Guidel...
github.com

It's giving some errors. Please look at it and suggest the resolution.

Thanks.
Rahul.



Thanks.

Rahul Singh



________________________________

**Comment by qiaozha** [comment]
@rahuls-microsoft  Did you make sure azdev test <your-module-name> --live passes before you submit the PR?

**Comment by qiaozha** [comment]
@rahuls-microsoft  I recommend you to follow the steps here. https://github.com/Azure/autorest.az#how-to-use-azure-cli-code-generator 
![image](https://user-images.githubusercontent.com/55688292/113110850-22878f80-923a-11eb-878b-c271ffbd72d4.png)
